File: works/tmp/2166.cpp
#include <iostream>
#include <cmath>
using namespace std;

double cord[10001][2];

double CCW(double x1, double y1, double x2, double y2, double x3, double y3)
{
    double a = x1*y2 + x2*y3 + x3*y1;
    double b = x2*y1 + x3*y2 + x1*y3;

    return (a-b)/2;
//    if(a-b < 0) return ((a-b) * -1)/2;
//    else return (a-b)/2;
}

int main()
{
    int n; cin >> n;
    for(int i = 0; i < n; i++) cin >> cord[i][0] >> cord[i][1];
    double sum = 0;
    // 점 하나를 기준으로 삼각형들로 나눔
    for(int i = 1; i < n-1; i++)
    {
        sum += CCW(cord[0][0], cord[0][1], cord[i][0], cord[i][1], cord[i+1][0], cord[i+1][1]);
    }

    printf("%.1f", abs(sum));
}

File: works/tmp/2166.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.nio.Buffer
import java.util.*

fun CCW(x1:Double, y1:Double, x2:Double, y2:Double, x3:Double, y3:Double): Double
{
    val a = x1*y2 + x2*y3 + x3*y1
    val b = x2*y1 + x3*y2 + x1*y3
    return (a-b)/2
}

fun main()
{
    val br = BufferedReader(InputStreamReader(System.`in`))
    var cord = Array(10001) { Array<Double>(2) {0.0} }
    val n = br.readLine().toInt()
    for(i in 0 until n)
    {
        val _s = br.readLine().split(' ')
        cord[i][0] = _s[0].toDouble()
        cord[i][1] = _s[1].toDouble()
    }

    var sum = 0.0
    for(i in 1 until n-1)
    {
        sum += CCW(cord[0][0], cord[0][1], cord[i][0], cord[i][1], cord[i+1][0], cord[i+1][1]);
    }

    System.out.format("%.1f", Math.abs(sum))
}